src/main/java/com/cburch/logisim/analyze/model/TruthTable.java

```diff
@@ -429,14 +429,30 @@ public void setVisibleRows(List<Entry[]> newEntries, boolean force) {
       final var r = newRows.get(i);
       for (final var idx : r) {
         if (taken[idx] != 0 && !force) {
-          throw new IllegalArgumentException(
+          boolean outputsEqual = true;
+          final var existingValues = newEntries.get(taken[idx] - 1);
+          final var currentValues = newEntries.get(i);
+          for (int col = 0; col < no; col++) {
+            final var existingValue = existingValues[ni + col];
+            final var currentValue = currentValues[ni + col];
+            if (existingValue == Entry.DONT_CARE || currentValue == Entry.DONT_CARE) {
+              continue;
+            }
```

Comment:
I think it would be simpler to eliminate the outputsEqual variable and change the test to:

`if (existingValue != Entry.DONT_CARE && currentValue != Entry.DONT_CARE) {
  if (!currentValue.equals(existingValue) {
      throw IllegaleArgumentException(...`

Then delete the declaration of outputsEqual.

Reply:
clean!!